### PR TITLE
pass sources_for_koji_build_id to sources build

### PR DESCRIPTION
**Pull request by rcerven**
OSBS-8041

Signed-off-by: Robert Cerven <rcerven@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [ ] Pull request includes link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: osbs/build/user_params.py [hunk 1]
```diff
@@ -419,6 +419,7 @@ def __init__(self, build_json_dir=None):
         super(SourceContainerUserParams, self).__init__(
             build_json_dir=build_json_dir)
         self.sources_for_koji_build_nvr = BuildParam("sources_for_koji_build_nvr")
+        self.sources_for_koji_build_id = BuildParam("sources_for_koji_build_id")
```
Comment (MartinBasti) [inline osbs/build/user_params.py]: IMHO it should have `allow_none=True`

**Comment by MartinBasti** [comment]
Shall be CLI integration of new option be part of this PR?

File: tests/build_/test_user_params.py [hunk 2]
```diff
@@ -392,11 +392,11 @@ def test_validate_missing_required(self):
         with pytest.raises(OsbsValidationException):
             spec.validate()
 
-    @pytest.mark.parametrize('origin_nvr, final_name', [
-        ('test-1-123', 'test-source'),
-        ('test-dashed-nvr-1-123', 'test-dashed-nvr-source'),
+    @pytest.mark.parametrize('origin_nvr, final_name, origin_id', [
+        ('test-1-123', 'test-source', 12345),
+        ('test-dashed-nvr-1-123', 'test-dashed-nvr-source', 12345),
```
Comment (twaugh) [inline tests/build_/test_user_params.py]: Let's also have a test which does not pass in an ID.

File: osbs/build/user_params.py (hunk 1 above)
Comment (rcerven) [reply osbs/build/user_params.py]: sure

File: tests/build_/test_user_params.py (hunk 2 above)
Comment (rcerven) [reply tests/build_/test_user_params.py]: ok

**Comment by rcerven** [comment]
> Shall be CLI integration of new option part of this PR?

added

**Comment by rcerven** [comment]
rn not updated, fix for unreleased feature

**Pull request merged**